src/DFTK.jl

```diff
@@ -177,13 +177,19 @@ function __init__()
     # DoubleFloats has been loaded (via a "using" or an "import").
     # See https://github.com/JuliaPackaging/Requires.jl for details.
     #
-    # The global variable GENERIC_FFT_LOADED makes sure that things are only
-    # included once.
+    # The global variables GENERIC_FFT_LOADED and DUMMY_INPLACE_LOADED
+    # make sure that things are only included once.
+    @require ForwardDiff = "f6369f11-7733-5829-9624-2563aa707210" begin
+        !isdefined(DFTK, :DUMMY_INPLACE_LOADED) && include("workarounds/dummy_inplace_fft.jl")
+        include("workarounds/forwarddiff_rules.jl")
+    end
     @require IntervalArithmetic="d1acc4aa-44c8-5952-acd4-ba5d80a2a253" begin
         include("workarounds/intervals.jl")
+        !isdefined(DFTK, :DUMMY_INPLACE_LOADED) && include("workarounds/dummy_inplace_fft.jl")
```

Comment:
I'm confused, why is that needed here?

Reply:
It could happen someones uses ForwardDiff and DoubleFloats. So you don't want to include twice.

Reply:
Sure but this loads the forwarddiff stuff when intervalarithmetic is loaded?

Reply:
dummy inplace is also needed for generic ffts

Reply:
then can we just have include("dummy_inplace_fft.jl") in fft_generic.jl?

Reply:
dummy_inplace was previously part of generic_fft.jl (it does not include ForwardDiff stuff but is a common prerequisite for all three of ForwardDiff, DoubleFloats and IntervalArithmetic)

Reply:
Or not... never mind anyway

Reply:
we could, but I think that's only more confusing.

Reply:
Yup OK!